packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/routing/scripts/cdp-page-view.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { isServer, CdpHelper, LayoutServicePageState } from '@sitecore-jss/sitecore-jss-angular';
+import { pageView, PageViewData } from '@sitecore-cloudsdk/events/browser';
+import { JssContextService } from '../../jss-context.service';
+import { JssState } from '../../JssState';
+
+/**
+ * This is the CDP page view component.
+ * It uses the Sitecore Cloud SDK to enable page view events on the client-side.
+ * See Sitecore Cloud SDK documentation for details.
+ * https://www.npmjs.com/package/@sitecore-cloudsdk/events
+ */
+@Component({
+  selector: 'app-cdp-page-view',
+  template: '',
+})
+export class CdpPageViewComponent implements OnInit {
+  private contextSubscription: Subscription;
+
+  constructor(private jssContext: JssContextService) {}
+
+  ngOnInit(): void {
+    if (!isServer()) {
+      this.contextSubscription = this.jssContext.state.subscribe((newState: JssState) => {
+        const {
+          route,
+          context: { pageState, language, variantId },
+        } = newState.sitecore;
+
+        if (pageState !== LayoutServicePageState.Normal || !route?.itemId) {
+          return;
+        }
+
+        // Do not create events if disabled (e.g. we don't have consent)
+        if (this.disabled()) {
+          return;
+        }
+
+        const scope = process.env.ANGULAR_PUBLIC_PERSONALIZE_SCOPE;
+        const pageVariantId = CdpHelper.getPageVariantId(
+          route.itemId,
+          language,
```

Comment:
Looking at Next.js implementation and thinking that we need to handle "language" in a similar way, see [here](https://github.com/Sitecore/jss/blob/dev/packages/create-sitecore-jss/src/templates/nextjs-xmcloud/src/components/CdpPageView.tsx#L40)